coconut/configuration/configurationutil.go

```diff
@@ -0,0 +1,224 @@
+/*
+ * === This file is part of ALICE O² ===
+ *
+ * Copyright 2019 CERN and copyright holders of ALICE O².
+ * Author: George Raduta <george.raduta@cern.ch>
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * In applying this license CERN does not waive the privileges and
+ * immunities granted to it by virtue of its status as an
+ * Intergovernmental Organization or submit itself to any jurisdiction.
+ */
+
+package configuration
+
+import (
+	"github.com/fatih/color"
+	"github.com/olekukonko/tablewriter"
+	"github.com/spf13/cobra"
+	"strconv"
+	"time"
+	"io"
+	"io/ioutil"
+	"os"
+	"fmt"
+	"strings"
+	"errors"
+	"regexp"
+	"encoding/json"
+	"gopkg.in/yaml.v2"
+)
+
+var  (
+	inputFullRegex = regexp.MustCompile(`^([a-zA-Z0-9-]+)(\/[a-z-A-Z0-9-]+){1}(\@[0-9]+)?$`)
+)
+var(
+	blue = color.New(color.FgHiBlue).SprintFunc()
+	red = color.New(color.FgHiRed).SprintFunc()
+)
+
+func isInputCompEntryTsValid(input string) bool {
+	return inputFullRegex.MatchString(input)
+}
+
+func isInputSingleValidWord(input string) bool {
+	return !strings.Contains(input, "/") && !strings.Contains(input, "@")
+}
+
+// Method to parse a timestamp in the specified format
+func getTimestampInFormat(timestamp string, timeFormat string)(string, error){
+	timeStampAsInt, err := strconv.ParseInt(timestamp, 10, 64)
+	if err != nil {
+		return "", errors.New(fmt.Sprintf("Unable to identify timestamp"))
+	}
+	tm := time.Unix(timeStampAsInt, 0)
+	return  tm.Format(timeFormat), nil
+}
+
+// Method to return the latest timestamp for a specified component & entry
+// If no keys were passed an error and code exit 3 will be returned
+func getLatestTimestamp(keys []string, component string, entry string)(timestamp string, err error, code int) {
+	keyPrefix := componentsPath + component + "/" + entry
+	if len(keys) == 0 {
+		err = errors.New(fmt.Sprintf("No keys found"))
+		return "", err, emptyData
+	}
+
+	var maxTimeStamp uint64
+	for _, key := range keys {
+		componentTimestamp, err := strconv.ParseUint(strings.TrimPrefix(key, keyPrefix + "/"), 10, 64)
```

Comment:
I imagine this is done on purpose here, but this leads to variable shadowing, which could in turn lead to errors. A very good explanation can be found [here](https://rpeshkov.net/blog/golang-variable-shadowing/).

Reply:
In this case I am not interested in stopping the whole command if one of the timestamps is bad that is why I am adding it to the list of elements only if it is a valid one. So even if the timestamp is not correct, it will be ignored.

The error that I am interested here is the one with emptyData